conans/client/rest/rest_client_common.py

```diff
@@ -162,15 +162,15 @@ def upload_recipe(self, ref, files_to_upload, deleted, retry, retry_wait):
             self._remove_conanfile_files(ref, deleted)
 
     def get_recipe_snapshot(self, ref):
+        # this method is used only for UPLOADING, then it requires the credentials
```

Comment:
Probably it would be better to move it to the uploaded just before calling this method:
```

    def _recipe_files_to_upload(self, ref, policy, the_files, remote, remote_manifest):
        # Get the remote snapshot
        remote_snapshot = self._remote_manager.get_recipe_snapshot(ref, remote)
```

Reply:
I know it would require to make available that method in the `auth_manager` and `remote_manager`

Reply:
Done

Reply:
it just reads a bit weird:

```python
class ConanApiAuthManager(object):

    @input_credentials_if_unauthorized
    def check_credentials(self):
        self._rest_client.check_credentials()
```

We might need to revise this pattern sometime, but not urgent

Reply:
Absolutely agree.